include/ginkgo/core/matrix/dense.hpp

```diff
@@ -694,6 +741,20 @@ class Dense
             make_temporary_output_clone(exec, result).get());
     }
 
+    /**
+     * Overload for scalar result, see @ref compute_norm2
+     * Assumes that the result is placed in host memory
+     */
+    void compute_norm2(ValueType *result) const
+    {
+        GKO_ASSERT_EQUAL_COLS(this, dim<2>(1, 1));
+        auto exec = this->get_executor();
+        auto dense_result = Dense::create(
+            exec->get_master(), dim<2>{1, 1},
+            Array<ValueType>::view(exec->get_master(), 1, result), 1);
+        this->compute_norm2(dense_result.get());
+    }
```

Comment:
I'm not sure about this one (same for dot): So far, I think we don't have any places where we pass pointers into the public interface except to initialize arrays. Without any information associated to it, a pointer can be on the host or a device, and we have no way to differentiate between them. Users might try to pass a device pointer to this function by accident?
When we have Frobenius norm calculation in there, we can definitely add an overload that returns a single value, but for now, I think this might be a step in the wrong direction.

Reply:
You're right, that overload (and the one for dot) is nor really fitting. I will remove it. Perhaps as part of #796 the overload with a single return value can be added.